libavfilter/qsvvpp.c

```diff
@@ -867,13 +876,33 @@ int ff_qsvvpp_init(AVFilterContext *avctx, QSVVPPParam *param)
 static int qsvvpp_init_vpp_session(AVFilterContext *avctx, QSVVPPContext *s, const QSVFrame *in, QSVFrame *out)
 {
     int ret;
+    mfxExtBuffer *ext_param[QSVVPP_MAX_FRAME_EXTBUFS];
+    QSVVPPFrameParam fp = { 0, ext_param };
 
     if (s->vpp_initted)
         return 0;
 
     s->vpp_param.vpp.In.PicStruct = in->surface.Info.PicStruct;
     s->vpp_param.vpp.Out.PicStruct = out->surface.Info.PicStruct;
 
+    ret = s->set_frame_ext_params(avctx, in->frame, out->frame, &fp);
```

Comment:
How about moving `if (s->vpp_initted)` check after calling `set_frame_ext_params()`? So that there is a chance to reinitialize VPP by resetting `vpp_initted` in function `set_frame_ext_params()` .

Reply:
Thanks for suggestion, this is in my TODO to  call MFXVideoVPP_Reset to reset a session for vpp

Reply:
Ok, Got it.